src/architect/project/project.utils.ts

```diff
@@ -0,0 +1,319 @@
+      if (!(Object.values(type_map).includes(type_name))) {
+        const dependency = dependency_map[component_spec.name] ? dependency_map[component_spec.name] : '';
+        
+        if (!component_spec.architect_yml) {
+          selections[component_spec.name.toLowerCase()] = {
+            name: component_spec.name,
+            architect_yml: component_spec,
+            depends_on: dependency,
+          };
+        } else {
+          selections[component_spec.name.toLowerCase()] = {
+            name: component_spec.name,
+            architect_yml: yaml.load(component_spec.architect_yml.join('\n')) as Dictionary<any>,
+            depends_on: dependency,
+          };
+        }
+      }
+    }
+
+    return selections;
+  }
+
+  static async prompt(choices: any[], message: string): Promise<any> {
+    inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
+    const answers: { selected: any } = await inquirer.prompt([
+      {
+        type: 'autocomplete',
+        name: 'selected',
+        message: message,
+        source: async (answers_so_far: any, input: string) => {
+          return choices.map((p) => ({ name: p.name, value: p }));
+        },
+      },
+    ]);
+    return answers.selected;
+  }
+
+  static async downloadRepo(url: string, dir_path: string): Promise<void> {
+    await download(url + `#${GITHUB_BRANCH}`, dir_path, (err: any) => {
+      if (err) {
+        throw new Error(`Failed to download repository ${url}`);
+      }
+    });
+  }
+
+  static async downloadGitHubRepos(selections: Dictionary<Dictionary<any>>, project_dir: string): Promise<void> {
+    // download any selection that has a homepage
+    for (const selection of Object.values(selections)) {
+      if (selection.architect_yml && selection.architect_yml.homepage) {
+        const url = selection.architect_yml.homepage.replace('https://github.com/', '');
+        await this.downloadRepo(url, project_dir + '/' + selection.name.toLowerCase());
+      }
+    }
+    await new Promise(resolve => setTimeout(resolve, 2000));
+  }
+
+  static async createArchitectYaml(selections: Dictionary<any>, project_dir: string): Promise<void> {
+    // get actual service name for depends_on
+    for (const selection of Object.values(selections)) {
+      if (selection.depends_on) {
+        const depends_on_proj = selections[selection.depends_on];
+        const services = depends_on_proj.architect_yml.services;
+        if (Object.keys(services).length === 1) {
+          selection.depends_on = Object.keys(services)[0];
+        } else {
+          const service_name = Object.keys(services).find((name: any) => !services[name].image);
+          selection.depends_on = service_name;
+        }
+      }
+    }
+
+    let interfaces: Dictionary<any> = {};
+    let dependencies: Dictionary<any> = {};
+    let secrets: Dictionary<any> = {};
+    let services: Dictionary<any> = {};
```

Comment:
These should be determined generically based on the loaded architect.yml files. For instance, if we want to showcase `tasks` this code will not work.